### ISSUE TITLE
 Lock down sharing level for site/role/user

### ISSUE THREAD

**Issue by marcnil815** [MISP/MISP#5909]
**Is your feature request related to a problem? Please describe.**
I am running a MISP with a wide user base. No Event must be shared above "This Community", however, users make mistakes and share to "Connected" and "All*"

**Describe the solution you'd like**
I would like the ability to lock down the possible sharing levels pr. site/role/user.
For now site/server would suffice, however role would be a much better implementation. User is nice to have for edge cases.

**Describe alternatives you've considered**
Having select gatekeepers with Publish rights (not viable as everyone wants to share)

Creating scripts that catch sharing and downgrade the sharing level if above "This Community" (not viable: race condition).

Or alter the View code to not display the selections (not viable....not touching the base code :))

**Additional context**
This might be an edge case, but for multi-tenant MISP hosting where Publish Gatekeepers are not viable (even they might make mistakes), this would be a large bump up the security/privacy scale (least privilege and all that).

Br
Marc

**Comment by mokaddem** [comment]
Hey @marcnil815, I totally get the feature request but I don't really see why you have an issue for your use case.
Is your instance connected to other MISP instances?
- If yes, can't you just disable the PUSH for theses instances? Because by enforcing `This Community`, nothing will ever by pushed anyway.
- If no, the distribution above `This Community` will not have any other effect.
Thanks

**Comment by marcnil815** [comment]
Other MISPs will still be able to PULL "Connected" and "All*" regardless of my PUSH settings.

Unless they elect not to do so on the receiving end; not that I am sure that is possible....you can't force degrade sharing levels on a PULL except "do not publish". And you would still extend the perimeter of sharing level to the next HUB. So even if it where possible I am depending on the other MISP to implement correctly.

We are not connected to other instance because of precisely this problem (or not precisely, but this is a mjor reason) :)

Hope this makes sense.

(edit: typos)

**Comment by mokaddem** [comment]
> Unless they elect not to do so on the receiving end; not that I am sure that is possible....you can't force degrade sharing levels on a PULL except "do not publish". And you would still extend the perimeter of sharing level to the next HUB. So even if it where possible I am depending on the other MISP to implement correctly

@marcnil815 That's totally true!
Considering the requirements you gave, I assumed that your users doesn't have an interconnected MISP instance with third parties. Also I found it weird that you want to prevent your users to share with others.
Is there a particular reason for that? Because if they really want to, they can still export the data, change the distribution level then publish it to their partners. Of course it introduces an small overhead but it's doable for them.

**Comment by marcnil815** [comment]
@mokaddem 
Correct: the MISP is not connected to other MISPs for the sole reason that sharing levels cannot be locked down. The terms of engagement is "one-hop-only". And there is no way to enforce that (.ie. "max this community only"). That's about it :)

We are well aware that anyone with ill will can do whatever they want with the data. The puporse of this feature request is to prevent _unintentional_ oversharing.

Think of it like this: the members want to prevent sharing outside of the group, but still want to connect their own MISP (those of the members who have one).

In before "Sharing Group": same concern. "unintentional" oversharing.

(edit: clarity)

**Comment by marcnil815** [comment]
Linked issue: #3394
